File: Data Structures & Algorithms/is-palindrome/submission-2.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int len=s.length();
        int left=0;
        int right=len-1;
        while(left<right){
            while(left<right && !isalnum(s[left])){
                left++;
            }
            while(left<right && !isalnum(s[right])){
                right--;
            }
            if(tolower(s[left])!=tolower(s[right])) return false;
            left++;
            right--;
        }
        return true;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-2.java
class Solution{
    public int maxArea(int[] heights){
        int best=0;
        int left=0;
        int right=heights.length-1;
        int area;
        while(left<right){
            area=Math.min(heights[left],heights[right])*(right-left);
            best=Math.max(best,area);
            if(heights[left]<heights[right]){
                left++;
            }else{
                right--;
            }
        }
        return best;
    }
}